webdriver_wharf/app.py

```diff
@@ -172,18 +183,21 @@ def container_info(container):
     host = requesting_host()
     host_noport = host.split(':')[0]
 
-    return {
+    container_info = {
         'image_id': container.image_id,
         'checked_out': container.checked_out,
         'checkin_url': 'http://%s/checkin/%s' % (host, container.name),
         'renew_url': 'http://%s/renew/%s' % (host, container.name),
         'webdriver_port': container.webdriver_port,
         'webdriver_url': 'http://%s:%d/wd/hub' % (host_noport, container.webdriver_port),
-        'vnc_port': container.vnc_port,
-        'vnc_display': 'vnc://%s:%d' % (host_noport, container.vnc_port - 5900),
-        'http_port': container.http_port,
-        'fileviewer_url': 'http://%s:%d/' % (host_noport, container.http_port),
     }
+    if vnc_enabled:
+        container_info['vnc_port'] = container.vnc_port
+        container_info['vnc_display'] = 'vnc://%s:%d' % (host_noport, container.vnc_port - 5900)
```

Comment:
Removing these from the container info is a backward-incompatible API change. While the API isn't well-defined, the docs (index page) at least guarantee that you will get a value for vnc_port and vnc_display, so they should be present in the container info, even if they're None.

I don't think this applies to `http_port`/`fileviewer_url`, since those are undocumented and at the moment I don't even think they work.